### DISCUSSION TITLE
val submit ok ,but testset submit code error

### DISCUSSION THREAD

**Discussion by CarlHuangNuc** [bdd100k/bdd100k#228] [General]
Hi,

         I do not know why validation submit can pass,

         testset submit always failed . Whether your server check have some issue.

Detail log:

  0%|          | 0/79636 [00:00<?, ?it/s]
  4%|��         | 3101/79636 [00:00<00:02, 31001.53it/s]
  8%|��         | 6320/79636 [00:00<00:02, 31345.84it/s]
 12%|����        | 9512/79636 [00:00<00:02, 31513.73it/s]
 16%|����        | 12896/79636 [00:00<00:02, 32177.14it/s]
 20%|����        | 15992/79636 [00:00<00:02, 31800.02it/s]
 24%|������       | 18866/79636 [00:00<00:01, 30812.65it/s]
 29%|������       | 22883/79636 [00:00<00:01, 33125.91it/s]
 33%|��������      | 26076/79636 [00:00<00:01, 32753.95it/s]
 37%|��������      | 29213/79636 [00:00<00:01, 31158.08it/s]
 40%|��������      | 32248/79636 [00:01<00:01, 29887.17it/s]
 45%|����������     | 35996/79636 [00:01<00:01, 31819.56it/s]
 49%|����������     | 39181/79636 [00:01<00:01, 31425.81it/s]
 53%|������������    | 42570/79636 [00:01<00:01, 32126.44it/s]
 57%|������������    | 45789/79636 [00:01<00:01, 31502.34it/s]
 62%|��������������   | 49662/79636 [00:01<00:00, 33369.96it/s]
 67%|��������������   | 53034/79636 [00:01<00:00, 32179.84it/s]
 71%|��������������   | 56530/79636 [00:01<00:00, 32965.25it/s]
 76%|����������������  | 60290/79636 [00:01<00:00, 34230.30it/s]
 80%|����������������  | 63746/79636 [00:01<00:00, 33362.38it/s]
 84%|������������������ | 67110/79636 [00:02<00:00, 33028.53it/s]
 88%|������������������ | 70433/79636 [00:02<00:00, 32135.73it/s]
 93%|��������������������| 73714/79636 [00:02<00:00, 32329.60it/s]
 97%|��������������������| 76961/79636 [00:02<00:00, 31262.40it/s]
100%|��������������������| 79636/79636 [00:02<00:00, 32247.34it/s]

  0%|          | 0/79636 [00:00<?, ?it/s]
  4%|��         | 2803/79636 [00:00<00:02, 28024.48it/s]
  7%|��         | 5660/79636 [00:00<00:02, 28185.49it/s]
 11%|��         | 8519/79636 [00:00<00:02, 28304.98it/s]
 14%|����        | 11438/79636 [00:00<00:02, 28564.57it/s]
 18%|����        | 14290/79636 [00:00<00:02, 28550.13it/s]
 22%|������       | 17239/79636 [00:00<00:02, 28823.79it/s]
 25%|������       | 20111/79636 [00:00<00:02, 28791.49it/s]
 29%|������       | 23049/79636 [00:00<00:01, 28963.84it/s]
 33%|��������      | 25935/79636 [00:00<00:01, 28931.27it/s]
 36%|��������      | 28807/79636 [00:01<00:01, 28866.97it/s]
 40%|��������      | 31704/79636 [00:01<00:01, 28897.05it/s]
 43%|����������     | 34539/79636 [00:01<00:01, 28619.13it/s]
 47%|����������     | 37389/79636 [00:01<00:01, 28580.53it/s]
 51%|����������     | 40244/79636 [00:01<00:01, 28571.29it/s]
 54%|������������    | 43162/79636 [00:01<00:01, 28750.09it/s]
 58%|������������    | 46061/79636 [00:01<00:01, 28820.90it/s]
 61%|��������������   | 48935/79636 [00:01<00:01, 28607.68it/s]
 65%|��������������   | 51790/79636 [00:01<00:00, 28567.10it/s]
 69%|��������������   | 54722/79636 [00:01<00:00, 28788.36it/s]
 72%|����������������  | 57599/79636 [00:02<00:00, 28642.58it/s]
 76%|����������������  | 60462/79636 [00:02<00:00, 28567.20it/s]
 80%|����������������  | 63318/79636 [00:02<00:00, 28429.37it/s]
 83%|������������������ | 66161/79636 [00:02<00:00, 28280.09it/s]
 87%|������������������ | 68989/79636 [00:02<00:00, 28230.84it/s]
 90%|������������������ | 71812/79636 [00:02<00:00, 28111.58it/s]
 94%|��������������������| 74648/79636 [00:02<00:00, 28178.01it/s]
 97%|��������������������| 77496/79636 [00:02<00:00, 28266.29it/s]
100%|��������������������| 79636/79636 [00:02<00:00, 28561.78it/s]
Traceback (most recent call last):
  File "/code/scripts/workers/submission_worker.py", line 491, in run_submission
    submission_metadata=submission_serializer.data,
  File "/tmp/tmp9w9u77pr/compute/challenge_data/challenge_1653/main.py", line 91, in evaluate
    nproc=1,
  File "/usr/local/lib/python3.7/site-packages/scalabel/eval/mot.py", line 355, in evaluate_track
    assert len(gts) == len(results)
AssertionError

**Comment by CarlHuangNuc** [comment]
![image](https://user-images.githubusercontent.com/104048874/164373580-da632c67-cab7-4601-8173-34586aecb79d.png)

**Comment by CarlHuangNuc** [comment]
the length of testset is not 79636???

how many is  len(gts) ？？

**Comment by thomasehuang** [comment]
From the output it seems like the length of gts and results are the same. Can you check whether the videoName field for each frame in your prediction file is set correctly? It should be set to the name of the video rather than the image.
